### PR TITLE
RF: mostly fixes for Py3 compatibility and new dependency auto-install

### PR DESCRIPTION
**Pull request by peircej**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -24,10 +24,11 @@
             'pyosf', 'requests[security]',
             'xlrd', 'openpyxl',
             'pyyaml', 'gevent', 'msgpack-python', 'psutil', 'tables'
+            'opencv-python',
             ]
 # some optional dependencies
 if platform == 'darwin':
-    required.append('pyobjc-core')
+    required.append('pyobjc-core', 'pyobjc-framework-Quartz')
```
Comment (kaczmarj) [inline setup.py]: This raises a TypeError. I suggest `required.extend(['pyobjc-core', 'pyobjc-framework-Quartz'])`

Comment (peircej) [inline setup.py]: Correct. My bad. Do you want to submit that?

Comment (kaczmarj) [inline setup.py]: Sorry, I'm a newbie at reviews. Do you mean submit a review and request changes? Or do you want me to submit a PR to your branch?

Comment (peircej) [inline setup.py]: I was meaning submit a PR (I like the originator of fixes and features to get credit for their contributions). I'll pull the branch in and then you can do the fix in the main master (or I'm perfectly happy to if you prefer)

Comment (kaczmarj) [inline setup.py]: Thanks! Working on it now.

**Pull request merged**